cabal-install/Distribution/Client/CmdShowBuildInfo.hs

```diff
@@ -0,0 +1,254 @@
+     ++ "'cabal.project.local' and other files.",
+  commandNotes        = Just $ \pname ->
+        "Examples:\n"
+     ++ "  " ++ pname ++ " new-build\n"
+     ++ "    Build the package in the current directory or all packages in the project\n"
+     ++ "  " ++ pname ++ " new-build pkgname\n"
+     ++ "    Build the package named pkgname in the project\n"
+     ++ "  " ++ pname ++ " new-build ./pkgfoo\n"
+     ++ "    Build the package in the ./pkgfoo directory\n"
+     ++ "  " ++ pname ++ " new-build cname\n"
+     ++ "    Build the component named cname module Distribution.Client.InstallPlanin the project\n"
+     ++ "  " ++ pname ++ " new-build cname --module Distribution.Client.InstallPlanenable-profiling\n"
+     ++ "    Build the component in profilingmodule Distribution.Client.InstallPlan mode (including dependencies as needed)\n\n"
+
+     ++ cmdCommonHelpTextNewBuildBeta
+   }
+
+
+-- | The @build@ command does a lot. It brings the install plan up to date,
+-- selects that part of the plan needed by the given or implicit targets and
+-- then executes the plan.
+--
+-- For more details on how this works, see the module
+-- "Distribution.Client.ProjectOrchestration"
+--
+showBuildInfoAction :: (ConfigFlags, ConfigExFlags, InstallFlags, HaddockFlags, TestFlags)
+            -> [String] -> GlobalFlags -> IO ()
+showBuildInfoAction (configFlags, configExFlags, installFlags, haddockFlags, testFlags)
+            targetStrings globalFlags = do
+
+  baseCtx <- establishProjectBaseContext verbosity cliConfig
+  let baseCtx' = baseCtx {
+                    buildSettings = (buildSettings baseCtx) {
+                      buildSettingDryRun = True
+                    }
+                  }
+
+  targetSelectors <- either (reportTargetSelectorProblems verbosity) return
+                  =<< readTargetSelectors (localPackages baseCtx') Nothing targetStrings
+
+  buildCtx <-
+    runProjectPreBuildPhase verbosity baseCtx' $ \elaboratedPlan -> do
+      -- Interpret the targets on the command line as build targets
+      -- (as opposed to say repl or haddock targets).
+      targets <- either (reportTargetProblems verbosity) return
+                $ resolveTargets
+                    selectPackageTargets
+                    selectComponentTarget
+                    TargetProblemCommon
+                    elaboratedPlan
+                    Nothing
+                    targetSelectors
+
+      -- Don't prune the plan though, as we want a list of all configured packages
+      return (elaboratedPlan, targets)
+
+  scriptLock <- newLock
+  showTargets verbosity baseCtx' buildCtx scriptLock
+  
+  where
+    -- Default to silent verbosity otherwise it will pollute our json output
+    verbosity = fromFlagOrDefault silent (configVerbosity configFlags)
+    cliConfig = commandLineFlagsToProjectConfig
+                  globalFlags configFlags configExFlags
+                  installFlags defaultClientInstallFlags
```

Comment:
The function  `commandLineFlagsToProjectConfig` has changed signature and I was not able to determine the fitting parameter for `ClientInstallFlags`, thus, I used the `defaultClientInstallFlags`.
Do we have to honor clientInstallFlags?

Reply:
Offhand - yes, clientInstallFlags should be honored, otherwise what would normally install on the user machine/configuration would fail to install because of missing flags (such as pointing to libraries living in a non-usual place)

Reply:
Offhand - yes, clientInstallFlags should be honored, otherwise what would normally install on the user machine/configuration would fail to install because of missing flags (such as pointing to libraries living in a non-usual place)

Reply:
Where do I get the clientInstallFlags?

Reply:
Alas, I don't know. But I have been bitten by the problem of stack losing the flags down the invocation/dependencies chain and resetting them to defaults, which broke installation for me. So, naturally, I'm trying to minimize the chances that problem would bite me again.

Reply:
Actually, I dont think they matter, since we do not actually install anything?